com.unity.netcode.gameobjects/Runtime/NetworkVariable/Collections/NetworkDictionary.cs

```diff
@@ -9,9 +10,9 @@ namespace Unity.Netcode
     /// </summary>
     /// <typeparam name="TKey">The type for the dictionary keys</typeparam>
     /// <typeparam name="TValue">The type for the dictionary values</typeparam>
-    public class NetworkDictionary<TKey, TValue> : NetworkVariableBase, IDictionary<TKey, TValue> where TKey : unmanaged where TValue : unmanaged
+    public class NetworkDictionary<TKey, TValue> : NetworkVariableBase where TKey : unmanaged, IEquatable<TKey> where TValue : unmanaged
     {
-        private readonly IDictionary<TKey, TValue> m_Dictionary = new Dictionary<TKey, TValue>();
+        private NativeHashMap<TKey, TValue> m_Dictionary = new NativeHashMap<TKey, TValue>(64, Allocator.Persistent);
```

Comment:
Pardon my ignorance but what is the main advantage to use a NativeHashMap? Performances?

Reply:
Burst & Jobs friendliness and our future plans to piggyback on Unity.Collections instead of .NET collections etc.
Reasons does include performance and other things but if you know about Burst+Jobs+Collections combo, you already know what I mean :)